File: PracticeAndroid/AndroidPractice2/chatapp/LifeCycle/app/src/main/java/com/example/lifecycle/MainActivity.java
package com.example.lifecycle;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d("lifecycle", "oncreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("lifecycle", "onstart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("lifecycle", "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("lifecycle", "onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("lifecycle", "onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("lifecycle", "onDestroy");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d("lifecycle", "onRestart");
    }
}

File: PracticeAndroid/Andorid_Kotlin/ADExpress/RoomDB/app/src/main/java/com/androdevdk/roomdb/MainActivity.kt
package com.androdevdk.roomdb

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import com.androdevdk.roomdb.database.ContactDatabase
import com.androdevdk.roomdb.entity.Contact
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.util.*

class MainActivity : AppCompatActivity() {
    private val TAG = "MainActivity"
    lateinit var database: ContactDatabase
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        database = ContactDatabase.getDatabase(this)
        // val database1 = ContactDatabase.getDatabase(this)
//        database=Room.databaseBuilder(applicationContext,
//        ContactDatabase::class.java,
//        "contactDB").build()

        GlobalScope.launch {
            // database.contactDao().insertContact(Contact(0, "dk", "1234567890"))

            var contactData: MutableList<Contact> = mutableListOf<Contact>()
            contactData.add(Contact(0, "dk1", "1234567890"))
            contactData.add(Contact(0, "dk2", "1234567890"))
            contactData.add(Contact(0, "dk4", "1234567890"))
            database.contactDao().insertMultipleContacts(contactData)
            Log.d(TAG, "onCreate: " + "Data inserted successfully...")
        }
    }

    fun getData(view: View) {
        database.contactDao().getListOfContact().observe(this, androidx.lifecycle.Observer {
            Log.d("TAG", "getData: " + it.toString())
        })
    }
}